manipmongo/compiler.go

```diff
@@ -151,14 +133,14 @@ func CompileFilter(f *elemental.Filter) bson.D {
 		case elemental.AndFilterOperator:
 			subs := []bson.D{}
 			for _, sub := range f.AndFilters()[i] {
-				subs = append(subs, CompileFilter(sub))
+				subs = append(subs, CompileFilter(sub, opts...))
 			}
 			ands = append(ands, bson.D{{Name: "$and", Value: subs}})
 
 		case elemental.OrFilterOperator:
 			subs := []bson.D{}
 			for _, sub := range f.OrFilters()[i] {
-				subs = append(subs, CompileFilter(sub))
+				subs = append(subs, CompileFilter(sub, opts...))
```

Comment:
The new unit tests I added cover the recursive cases to ensure the configured options are passed down on recursive calls 🔁 

@primalmotion 👀